File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/configs/parameters/ConfigsCreateParameters.java
package co.uk.swarmbit.docker.api.configs.parameters;

import co.uk.swarmbit.docker.api.common.json.ConfigSpecJson;
import co.uk.swarmbit.docker.api.common.parameters.RequestBodyParameter;

public class ConfigsCreateParameters implements RequestBodyParameter {

    private ConfigSpecJson config;

    public ConfigsCreateParameters() {
        config = new ConfigSpecJson();
    }

    public ConfigsCreateParameters setConfig(ConfigSpecJson config) {
        this.config = config;
        return this;
    }

    @Override
    public Object getRequestBody() {
        return config;
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/configs/parameters/ConfigsCreateParameters.kt
package com.swarmbit.docker.api.configs.parameters

import com.swarmbit.docker.api.common.parameters.RequestBodyParameter
import com.swarmbit.docker.api.configs.model.ConfigSpec

class ConfigsCreateParameters : RequestBodyParameter {

    private var config: ConfigSpec = ConfigSpec()

    fun setConfig(config: ConfigSpec): ConfigsCreateParameters {
        this.config = config
        return this
    }

    override val requestBody: Any
        get() = config
}
